cupy/lib/polynomial.pyx

```diff
@@ -129,11 +129,20 @@ cdef class poly1d:
     def __pos__(self):
         return self
 
-    # TODO(Dahlia-Chehata): use polymul for non-scalars
     def __mul__(self, other):
+        if _should_use_rop(self, other):
+            return other.__rmul__(self)
         if cupy.isscalar(other):
+            # case: poly1d * python scalar
             return poly1d(self.coeffs * other)
-        raise NotImplementedError
+        if isinstance(self, numpy.generic):
+            # for consistency: numpy scalar * poly1d
```

Comment:
If you want to comment about the "consistency", more detail should be written. This comment does not mention about what is the consistency.